src/containers/Event.tsx

```diff
@@ -0,0 +1,50 @@
+import * as React from 'react';
+import {IChangeEvent} from 'react-jsonschema-form';
+import {EditableForm} from '../lib/cms/components/EditableForm';
+import {isUserAuthorized} from '../utils/auth';
+import {ICMSData} from '../models';
+
+interface IEventContext {
+  cmsData: ICMSData // CMS data to be rendered
+  onCmsDataUpdate: (data: any, path: string[]) => void // Callback to execute when a particular CMS
+                                                              // data needs to be updated
+}
+
+export class Event extends React.Component<any, any> {
+
+  context: IEventContext;
+
+  static contextTypes: React.ValidationMap<any> = {
+    cmsData: React.PropTypes.object.isRequired,
+    onCmsDataUpdate: React.PropTypes.func.isRequired
+  }
+
+  private getSchema(): any {
+    return {
+      title: 'Event Form',
+      type: 'object',
+      required: ['title', 'location'],
+      properties: {
+        title: {type: 'string', title: 'Title'},
+        location: {type: 'string', title: 'Location'}
+      }
+    };
+  }
+
+  public render(): JSX.Element {
+    const {cmsData} = this.context;
+    return (
+      <div>
+        <EditableForm isUserAuthorized={isUserAuthorized()}
+                      schema={this.getSchema()}
+                      formData={cmsData.event}
+                      onSubmit={(data: IChangeEvent) => {
```

Comment:
Do you think the EditableForm needs to expose this callback? Why don't we call the this.context.onCmsDataUpdate() from within the EditableForm component?